consensus/clique/clique.go

```diff
@@ -715,23 +730,45 @@ func CliqueRLP(header *types.Header) []byte {
 }
 
 func encodeSigHeader(w io.Writer, header *types.Header) {
-	err := rlp.Encode(w, []interface{}{
-		header.ParentHash,
-		header.UncleHash,
-		header.Coinbase,
-		header.Root,
-		header.TxHash,
-		header.ReceiptHash,
-		header.Bloom,
-		header.Difficulty,
-		header.Number,
-		header.GasLimit,
-		header.GasUsed,
-		header.Time,
-		header.Extra[:len(header.Extra)-crypto.SignatureLength], // Yes, this will panic if extra is too short
-		header.MixDigest,
-		header.Nonce,
-	})
+	var err error
+	if header.BaseFee == nil {
+		err = rlp.Encode(w, []interface{}{
+			header.ParentHash,
+			header.UncleHash,
+			header.Coinbase,
+			header.Root,
+			header.TxHash,
+			header.ReceiptHash,
+			header.Bloom,
+			header.Difficulty,
+			header.Number,
+			header.GasLimit,
+			header.GasUsed,
+			header.Time,
+			header.Extra[:len(header.Extra)-crypto.SignatureLength], // Yes, this will panic if extra is too short
+			header.MixDigest,
+			header.Nonce,
+		})
+	} else {
+		err = rlp.Encode(w, []interface{}{
+			header.ParentHash,
+			header.UncleHash,
+			header.Coinbase,
+			header.Root,
+			header.TxHash,
+			header.ReceiptHash,
+			header.Bloom,
+			header.Difficulty,
+			header.Number,
+			header.GasLimit,
+			header.GasUsed,
+			header.Time,
+			header.Extra[:len(header.Extra)-crypto.SignatureLength], // Yes, this will panic if extra is too short
+			header.MixDigest,
+			header.Nonce,
+			header.BaseFee,
+		})
+	}
```

Comment:
Can't this be reduced to:
```suggestion
         interfaces := []interface{}{
		header.ParentHash,
		header.UncleHash,
		header.Coinbase,
		header.Root,
		header.TxHash,
		header.ReceiptHash,
		header.Bloom,
		header.Difficulty,
		header.Number,
		header.GasLimit,
		header.GasUsed,
		header.Time,
		header.Extra[:len(header.Extra)-crypto.SignatureLength], // Yes, this will panic if extra is too short
		header.MixDigest,
		header.Nonce,
	}
	if header.BaseFee == nil {
		inferfaces = append(interfaces, header.BaseFee
	}
	if err = rlp.Encode(w, interfaces); err != nil {
	        panic("can't encode: " + err.Error())
	}
```

Reply:
Love the reductions/simplifications, will add these changes. There's a missing closing parenthesis on the `append` on line 752 so I can't commit the suggestions directly.